src/interactive/results.ts

```diff
@@ -351,14 +368,18 @@ export function removeAll(editor: vscode.TextEditor | null = null) {
 
 export function removeCurrent(editor: vscode.TextEditor) {
     editor.selections.forEach(selection => {
-        const isvalid = (result: Result) => {
-            const intersect = selection.intersection(result.range)
-            return result.document === editor.document && intersect !== undefined
-        }
-        results.filter(isvalid).forEach(removeResult)
+        results.filter(r => isResultInLineRange(editor, r, selection)).forEach(removeResult)
     })
 }
 
+function isResultInLineRange(editor: vscode.TextEditor, result: Result, range: vscode.Selection | vscode.Range) {
+    const intersect = range.intersection(result.range)
+    const lineRange = new vscode.Range(range.start.with(undefined, 0), editor.document.validatePosition(range.start.with(undefined, LINE_INF)))
+    const lineIntersect = lineRange.intersection(result.range)
+    console.log(result.document === editor.document, intersect, lineIntersect)
```

Comment:
So `range.start.with`'s `line` parameter will use that of `range`, right ?
I'm not sure exact cases where this pass gets hit though.

Reply:
Consider e.g. 
```
|    3+3
```
where you selected `3+3` and evaluated that. If you then place the cursor at the `|`, you can't clear the result with the current logic.

Reply:
I get it. Sounds nice.